lib/events.js

```diff
@@ -506,6 +507,7 @@ EventEmitter.prototype.emit = function emit(type, ...args) {
 
     // At least give some kind of context to the user
     const err = new ERR_UNHANDLED_ERROR(stringifiedEr);
+    err.name = 'UnhandledErrorEvent';
```

Comment:
the name usually corresponds with a class - instead of setting .name, what about subclassing whatever ERR_UNHANDLED_ERROR is? (TypeError or whatever) that’d allow instanceof to still work

Reply:
Yeap! That makes more sense, thanks!